spec/models/search/tag_query_spec.rb

```diff
@@ -1,186 +1,195 @@
+      tag_query = TagQuery.new(name: "abc-d")
+      results = tag_query.search_results
+      results.should include(tags[:fan_abc_d_minus])
+      results.should include(tags[:char_abc_d_minus])
+      results.should_not include(tags[:fan_abcd])
+      results.should_not include(tags[:char_abc])
+    end
+
+    it "preserves minus (-) preceded by a space ('abc -d' matches 'abc -d', 'abc d' and 'abc-d', but not 'abc')" do
+      tag_query = TagQuery.new(name: "abc -d")
+      results = tag_query.search_results
+      results.first.should eq(tags[:char_abc_d_minus])
+      results.should include(tags[:char_abc_d])
+      results.should include(tags[:fan_abc_d_minus])
+      results.should_not include(tags[:char_abc])
+    end
+
+    it "preserves slashes without quotes ('ab/cd' should match 'ab/cd' and 'ab cd')" do
+      tag_query = TagQuery.new(name: "ab/cd")
+      results = tag_query.search_results
+      results.should include(tags[:rel_slash])
+      results.should include(tags[:rel_space])
+    end
+
+    it "matches tags with canonical punctuation ('yuri!!!' on ice matches 'Yuri!!! On Ice')" do
+      tag_query = TagQuery.new(name: "yuri!!! on ice")
+      results = tag_query.search_results
+      results.should include(tags[:fan_yuri])
+    end
+
+    it "matches tags without canonical punctuation ('yuri on ice' matches 'Yuri!!! On Ice')" do
+      tag_query = TagQuery.new(name: "yuri on ice")
+      results = tag_query.search_results
+      results.should include(tags[:fan_yuri])
+    end
 
-  it "allows you to sort by Date Created in ascending order" do
-    q = TagQuery.new(sort_column: "created_at", sort_direction: "asc")
-    expect(q.generated_query[:sort]).to eq([{ "created_at" => { order: "asc", unmapped_type: "date" } }, { id: { order: "asc" } }])
+    it "matches unicode tags with unicode character ('Dave ♦ Sawbuck' matches 'Dave ♦ Sawbuck')" do
+      tag_query = TagQuery.new(name: "dave ♦ sawbuck")
+      results = tag_query.search_results
+      results.should include(tags[:rel_unicode])
+    end
+
+    it "matches unicode tags without unicode character ('dave sawbuck' matches 'Dave ♦ Sawbuck')" do
+      tag_query = TagQuery.new(name: "dave sawbuck")
+      results = tag_query.search_results
+      results.should include(tags[:rel_unicode])
+    end
+
+    it "defaults to TAGS_PER_SEARCH_PAGE to determine the number of results" do
+      allow(ArchiveConfig).to receive(:TAGS_PER_SEARCH_PAGE).and_return(5)
+      tag_query = TagQuery.new(name: "a*")
+      results = tag_query.search_results
+      expect(results.size).to eq 5
+    end
+
+    it "keeps sort order of tied tags the same when tag info is updated" do
+      tag_query = TagQuery.new(name: "abc*", sort_column: "created_at")
+      results = tag_query.search_results.map(&:id)
+
+      [tags[:fan_abcd], tags[:free_abccc]].each do |tag|
+        tag.update(canonical: true)
```

Comment:
Nitpick: I prefer to use `update!` in tests (although feel free to ignore this for now since it's not an added test)